package/Dockerfile

```diff
@@ -8,80 +12,91 @@ ENV CRIDOCKERD_VERSION=0.3.15
-COPY rke-etcd-backup /opt/rke-tools
+COPY templates /chroot/etc/confd/templates/
+COPY conf.d /chroot/etc/confd/conf.d/
+COPY cert-deployer nginx-proxy /chroot/usr/bin/
+COPY entrypoint.sh cloud-provider.sh weave-plugins-cni.sh /chroot/opt/rke-tools/
+COPY rke-etcd-backup /chroot/opt/rke-tools
 
-VOLUME /opt/rke-tools
-CMD ["/bin/bash"]
+ARG TARGETOS=linux
+ARG TARGETARCH=${ARCH}
 
-# Temporary image mostly to verify all binaries exist and are
-# valid for the target architecture.
-FROM tonistiigi/xx:1.4.0 AS xx
-FROM base as test
+# Verify key files that will be copied to the final
+# image exist and are statically linked to the target
+# architecture.
 COPY --from=xx / /
+RUN mkdir -p /run/lock
+RUN xx-verify --static /chroot/tmp/bandwidth \
+    && xx-verify --static /chroot/tmp/bridge \
+    && xx-verify --static /chroot/tmp/dhcp \
+    && xx-verify --static /chroot/tmp/firewall \
+    && xx-verify --static /chroot/tmp/flannel \
+    && xx-verify --static /chroot/tmp/host-device \
+    && xx-verify --static /chroot/tmp/host-local \
+    && xx-verify --static /chroot/tmp/ipvlan \
+    && xx-verify --static /chroot/tmp/loopback \
+    && xx-verify --static /chroot/tmp/macvlan \
+    && xx-verify --static /chroot/tmp/portmap \
+    && xx-verify --static /chroot/tmp/ptp \
+    && xx-verify --static /chroot/tmp/sbr \
+    && xx-verify --static /chroot/tmp/static \
+    && xx-verify --static /chroot/tmp/tuning \
+    && xx-verify --static /chroot/tmp/vlan \
+    && xx-verify --static /chroot/tmp/vrf
+
+RUN xx-verify --static /chroot/opt/rke-tools/bin/cri-dockerd \
+    && xx-verify --static /chroot/opt/rke-tools/bin/docker \
+    && xx-verify --static /chroot/opt/rke-tools/rke-etcd-backup
+
+RUN xx-verify --static /chroot/usr/bin/confd \
+    && xx-verify --static /chroot/usr/local/bin/kubectl
 
-ARG TARGETOS=linux
-ARG TARGETARCH=${ARCH}
 
-RUN xx-verify --static /tmp/bandwidth \
-    && xx-verify --static /tmp/bridge \
-    && xx-verify --static /tmp/dhcp \
-    && xx-verify --static /tmp/firewall \
-    && xx-verify --static /tmp/flannel \
-    && xx-verify --static /tmp/host-device \
-    && xx-verify --static /tmp/host-local \
-    && xx-verify --static /tmp/ipvlan \
-    && xx-verify --static /tmp/loopback \
-    && xx-verify --static /tmp/macvlan \
-    && xx-verify --static /tmp/portmap \
-    && xx-verify --static /tmp/ptp \
-    && xx-verify --static /tmp/sbr \
-    && xx-verify --static /tmp/static \
-    && xx-verify --static /tmp/tuning \
-    && xx-verify --static /tmp/vlan \
-    && xx-verify --static /tmp/vrf
-
-RUN xx-verify --static /opt/rke-tools/bin/cri-dockerd \
-    && xx-verify --static /opt/rke-tools/bin/docker \
-    && xx-verify --static /opt/rke-tools/rke-etcd-backup
-
-RUN xx-verify --static /usr/bin/confd \
-    && xx-verify --static /usr/local/bin/kubectl
-
-FROM base as final
+FROM scratch as final
+COPY --from=build /chroot /
+
+LABEL maintainer "Rancher Labs <support@rancher.com>"
```

Comment:
Wonder if we should change to SUSE Rancher instead.

Reply:
This PR is 4 months old now, and had to undergo a few rebases. If that's OK, I would rather avoid additional changes.